fix mul.Scalar with strided tensor

dolpm opened the pull request:
Summary: out variant has to be strided like self. since memory format isn't provided, this should be equivalent.

Test Plan:
prev. when we enable static dispatch this test would have numeric issues
```
buck2 test //caffe2/test:test_export -- test__scaled_dot_product_flash_attention_cpp_runtime_nonstrict --print-passing-details
```

Rollback Plan:

Reviewed By: SherlockNoMad

Differential Revision: D80191085

iremyux commented:
Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI and its test purposes - nothing about this PR specifically. (It should not effect the acceptance of the PR even if it fails.)

dolpm committed:
fix mul.Scalar with strided input (#160560)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/160560

out variant has to be strided like self. since memory format isn't provided, this should be equivalent.

Test Plan:
prev. when we enable static dispatch this test would have numeric issues
```
buck2 test //caffe2/test:test_export -- test__scaled_dot_product_flash_attention_cpp_runtime_nonstrict
```
```
buck2 test //sigmoid/backend/test:static_kernels_ops_test -- StaticKernelTest.MulScalar
```

Rollback Plan:

Reviewed By: SherlockNoMad

Differential Revision: D80191085

dolpm commented:
> Adding `ciflow/win-arm64` label to trigger Windows Arm64 CI and its test purposes - nothing about this PR specifically. (It should not effect the acceptance of the PR even if it fails.)

looks to be preventing this from being merged...

dolpm commented:
@pytorchmergebot merge -f "win arm failure shouldn't be related"
